qcodes/instrument_drivers/tektronix/DPO7200xx.py

```diff
@@ -0,0 +1,605 @@
+)
+
+from qcodes.utils.validators import Enum, Arrays
+
+
+def strip_quotes(string: str) -> str:
+    """
+    This function is used as a get_parser for various
+    parameters in this driver
+    """
+    return string.strip('"')
+
+
+class ModeError(Exception):
+    """
+    Raise this exception if we are in a wrong mode to
+    perform an action
+    """
+    pass
+
+
+class _TektronixDPOData(InstrumentChannel):
+    """
+    This is meant to be a private class, only to be used
+    by the TektronixDPO7000xx driver class. The end user
+    is not intended to call parameters on this submodule
+    directly.
+    """
+
+    def __init__(self, parent: Instrument, name: str) -> None:
+        super().__init__(parent, name)
+
+        self.add_parameter(
+            "start",
+            get_cmd="DATa:STARt?",
+            set_cmd="DATa:STARt {}",
+            get_parser=int
+        )
+
+        self.add_parameter(
+            "stop",
+            get_cmd="DATa:STOP?",
+            set_cmd="DATa:STOP {}",
+            get_parser=int
+        )
+
+        self.add_parameter(
+            "source",
+            get_cmd="DATa:SOU?",
+            set_cmd="DATa:SOU {}",
+            vals=Enum(*[
+                f"{name}{i}"
+                for name in TektronixDPOChannel.channel_types
+                for i in range(1, TektronixDPO7000xx.channel_count + 1)
+            ])
+        )
+
+        self.add_parameter(
+            "encoding",
+            get_cmd="DATa:ENCdg?",
+            set_cmd="DATa:ENCdg {}",
+            get_parser=strip_quotes,
+            vals=Enum(
+                "ASCIi",
+                "FAStest",
+                "RIBinary",
+                "RPBinary",
+                "FPBinary",
+                "SRIbinary",
+                "SRPbinary",
+                "SFPbinary",
+            ),
+            docstring="""
+            For a detailed explanation of the 
+            set arguments, please consult the 
+            programmers manual at page 263/264. 
+
+            http://download.tek.com/manual/077001022.pdf
+            """
+        )
+
+
+class _TektronixDPOWaveformFormat(InstrumentChannel):
+    """
+    This class abstracts the waveform formatting data
+    and is meant to be strictly private. Instances of
+    this class are coupled to specific channels, because
+    parameter values of this class depend on which
+    channel has been selected. The end user
+    is not intended to call parameters on this submodule
+    directly.
+    """
+
+    def __init__(self, parent: Instrument, name: str) -> None:
+        super().__init__(parent, name)
+
+        self.add_parameter(
+            "raw_data_offset",
+            get_cmd="WFMOutPRE:YOFF?",
+            get_parser=float,
+            docstring="""
+            Raw acquisition values range from min to max. 
+            For instance, for unsigned binary values of one 
+            byte, min=0 and max=255. The data offset specifies 
```

Comment:
does this mean that setting the min/max via a validator is left out of scope of this PR?

Reply:
No, its not out of the scope of this PR. The value of raw data offset is implicitly defined by several parameters, including the data encoding used (see the `encoding` parameter on TektronixDPOData), which in turn is determined by the data format (e.g. signed integer, floating points) and bytes per sample. See also the `TektronixDPOWaveformFormat` module. 

There is no way to directly set the value of `raw_data_offset` though.

Reply:
ok